core/services/s4/envelope.go

```diff
@@ -46,21 +49,43 @@ func (e Envelope) Sign(privateKey *ecdsa.PrivateKey) (signature []byte, err erro
 	return crypto.Sign(hash[:], privateKey)
 }
 
-// GetSignerAddress verifies the signature and returns the signing address.
-func (e Envelope) GetSignerAddress(signature []byte) (address common.Address, err error) {
+// Verifies the signature belongs to the sender's address
+func (e Envelope) VerifySignerAddress(signature []byte) (err error) {
 	if len(e.Address) != common.AddressLength {
-		return common.Address{}, fmt.Errorf("invalid address length: %d", len(e.Address))
+		return fmt.Errorf("invalid address length: %d", len(e.Address))
+	}
+
+	// Adjust the V component of the signature if using 27 or 28 instead of 0 or 1
```

Comment:
what is so important about 27 & 28? 

does the 64th elem have to be in [0,1,27,28]? if so, i would check that explicitly and return a nice error message or type. as is, if somehow it's not in that set, this will presumably always fail in a difficult-to-understand way

Reply:
The `crypto.SigToPub` provides error messages if this value is still incorrect after adjustment.

As for some context on why this value may need to be adjusted, see a similar Github issue [here](https://github.com/ethereum/go-ethereum/issues/19751).

Reply:
However, if you think this should be an explicit error stating that the `V` value of the signature is invalid, I can add.

Reply:
Looks like we need to repeat all this logic for Gateway too? @bolekk

Reply:
>As for some context on why this value may need to be adjusted, see a similar Github issue here.

maybe it's sufficient to link to that issue in a comment for anyone who reads the code and isn't familiar with these particular details

Reply:
Yes, just discussed with @KuphJr that we probably prefer to just always enforce the ETH prefix everywhere for consistency.